executor/grant_test.go

```diff
@@ -615,3 +615,17 @@ func TestGrantDynamicPrivs(t *testing.T) {
 	tk.MustQuery("SELECT Grant_Priv FROM mysql.user WHERE `Host` = '%' AND `User` = 'dyn'").Check(testkit.Rows("Y"))
 	tk.MustQuery("SELECT WITH_GRANT_OPTION FROM mysql.global_grants WHERE `Host` = '%' AND `User` = 'dyn' AND Priv='CONNECTION_ADMIN'").Check(testkit.Rows("Y"))
 }
+
+func TestNonExistTableIllegalGrant(t *testing.T) {
+	t.Parallel()
+
+	store, clean := testkit.CreateMockStore(t)
+	defer clean()
+
+	tk := testkit.NewTestKit(t, store)
+	tk.MustExec("create user i29302")
+	defer tk.MustExec("drop user i29302")
+
+	tk.MustGetErrCode("grant create temporary tables on tmpdb.tmp29302 to test_user", mysql.ErrIllegalGrantForTable)
```

Comment:
What is the purpose of this test? Do we know if tmpdb exists? Do we know if tmpdb.tmp29302 exists? (most likely the table does not exist at least)

I assume the purpose is to test Schema level grants (like create table) on table level? But the user and table name are a bit confusing, so it is hard to tell.